### PR TITLE
Added PAY and edited MOVE

### PR DESCRIPTION
**Pull request by saraalkuwari1**
#12

### CODE REVIEW

File: server/src/Account.java [hunk 1]
```diff
@@ -94,12 +94,14 @@ public String getAccountName() {
...
     }
 
-    public double getBalance() {
+    public static double getBalance() {
         return balance;
     }
 
+
+
     public void setBalance(double balance) {
```
Comment (gabrielbcn) [inline server/src/Account.java:103]: Why do you need to change this when it's not a static method?

File: server/src/NewBank.java [hunk 2]
```diff
@@ -202,6 +202,39 @@ private String deposit(Customer customer, String accountName,
         }
     }
 
+     /*
```
Comment (gabrielbcn) [inline server/src/NewBank.java:205]: All this is great. Simple and effective.

The only thing I did not understand is the change of this (dynamic) by Account (static) in the Account class. Thanks

File: server/src/Account.java [hunk 3]
```diff
@@ -71,7 +71,7 @@ public void setDate(String date) {
     private String accountId;
     private String userName;
     private String accountName;
-    private double balance;
+    private static double balance;
```
Comment (gabrielbcn) [inline server/src/Account.java:74]: balance cannot be static, it's related to a specific account not to all of them

File: server/src/Account.java [hunk 4]
```diff
@@ -94,12 +94,14 @@ public String getAccountName() {
         return accountName;
     }
 
-    public double getBalance() {
+    public static double getBalance() {
```
Comment (gabrielbcn) [inline server/src/Account.java:97]: Same with getbalance, there's one balance by account, not a single for all of them

File: server/src/Account.java [hunk 5]
```diff
@@ -112,16 +114,22 @@ public void setTransactions(List<Transaction> transactions) {
 
     public void addTransaction(Transaction transaction) {
         this.transactions.add(transaction);
-        this.balance = transaction.balance;
+        Account.balance = transaction.balance;
```
Comment (gabrielbcn) [inline server/src/Account.java:117]: Same as before, should be dynamic

File: server/src/Account.java [hunk 6]
```diff
@@ -112,16 +114,22 @@ public void setTransactions(List<Transaction> transactions) {
...
     }
 
+
     /*
      * Easily make a new transaction in this account
      */
     public void newTransaction(Double amount, String description) {
 
         this.addTransaction(new Account.Transaction(amount,
-                this.getBalance() + amount, description));
```
Comment (gabrielbcn) [inline server/src/Account.java:124]: This is a method that does a transaction in an account, now it seems to be doing two transactions within the same account. Not sure why.

File: server/src/Account.java [hunk 7]
```diff
@@ -130,7 +138,11 @@ public void newTransaction(Double amount, String description) {
     @Override
     public String toString() {
 
-        return "<" + this.userName + "#" + this.accountName + "#" + this.balance
+        return "<" + this.userName + "#" + this.accountName + "#" + Account.balance
                 + "#" + this.transactions + ">";
     }
+
+	public static Double getBalance(String accountName) {
```
Comment (gabrielbcn) [inline server/src/Account.java:145]: Same as before

File: server/src/NewBank.java (hunk 2 above)
Comment (saraalkuwari1) [reply server/src/NewBank.java:205]: My bad, sorry was trying different methods and I must did auto correct by mistake, will update and push

File: server/src/NewBank.java [hunk 8]
```diff
@@ -202,6 +202,39 @@ private String deposit(Customer customer, String accountName,
...
+        try {
+            amount = Double.parseDouble(amountString);
+        } catch (Exception e) {
+            return NewBank.FAIL;
+        }
+
+        Account account = customer.getAccountByName(accountName);
+        Account account1 = customer.getAccountByName(accountName1);
+
+        if (customer.hasAccountByName(accountName) && customer.hasAccountByName(accountName1) && amount > Account.getBalance(accountName)) {
```
Comment (NaomiOkiddy) [inline server/src/NewBank.java:222]: change
`amount > Account.getBalance(accountName)` to `amount > account.getBalance()`

**Pull request merged**